homeassistant/components/utility_meter/sensor.py

```diff
@@ -271,28 +275,30 @@ async def async_added_to_hass(self):
             self._last_reset = dt_util.parse_datetime(
                 state.attributes.get(ATTR_LAST_RESET)
             )
-            self.async_write_ha_state()
-            if state.attributes.get(ATTR_STATUS) == PAUSED:
-                # Fake cancellation function to init the meter paused
+            if state.attributes.get(ATTR_STATUS) == COLLECTING:
+                # Fake cancellation function to init the meter in similar state
                 self._collecting = lambda: None
 
         @callback
         def async_source_tracking(event):
             """Wait for source to be ready, then start meter."""
             if self._tariff_entity is not None:
-                _LOGGER.debug("Track %s", self._tariff_entity)
+                _LOGGER.debug(
+                    "<%s> tracks utility meter %s", self.name, self._tariff_entity
+                )
                 async_track_state_change_event(
                     self.hass, [self._tariff_entity], self.async_tariff_change
                 )
 
                 tariff_entity_state = self.hass.states.get(self._tariff_entity)
-                if self._tariff != tariff_entity_state.state:
-                    return
+                self._change_status(tariff_entity_state.state)
+                return
 
-            _LOGGER.debug("tracking source: %s", self._sensor_source_id)
+            _LOGGER.debug("<%s> collecting from %s", self.name, self._sensor_source_id)
             self._collecting = async_track_state_change_event(
                 self.hass, [self._sensor_source_id], self.async_reading
             )
+            self.async_write_ha_state()
```

Comment:
Why should we write state here? You've only added a state change listener.

Reply:
You are right! sorry